src/bp/core/routers/modules.ts

```diff
@@ -81,4 +176,30 @@ export class ModulesRouter extends CustomRouter {
       })
     )
   }
+
+  private async _getAllModules(): Promise<ModuleInfo[]> {
```

Comment:
These 2 functions seem a bit too complex to be in a router. Maybe extract them to a service?